yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -639,31 +655,44 @@ def run(self, info):
                 mp4_ass_warn = True
                 self.report_warning('ASS subtitles cannot be properly embedded in mp4 files; expect issues')
 
-        if not sub_langs:
+        if not sub_langs and not json_subs:
             return [], info
 
         input_files = [filename, *sub_filenames]
 
-        opts = [
-            *self.stream_copy_opts(ext=info['ext']),
-            # Don't copy the existing subtitles, we may be running the
-            # postprocessor a second time
-            '-map', '-0:s',
-        ]
+        opts = [*self.stream_copy_opts(ext=info['ext'])]
+
         for i, (lang, name) in enumerate(zip(sub_langs, sub_names)):
-            opts.extend(['-map', f'{i + 1}:0'])
             lang_code = ISO639Utils.short2long(lang) or lang
-            opts.extend([f'-metadata:s:s:{i}', f'language={lang_code}'])
+            opts.extend([
+                # Don't copy the existing subtitles, we may be running the
+                # postprocessor a second time
+                '-map', '-0:s',
```

Comment:
Why is this in the loop now? this will have it be passed multiple times right? `-map -0:s -map 1:0 -metadata:s:s:0 language=... -map -0:s -map 1:0`?